src/servers/src/hint_headers.rs

```diff
@@ -25,6 +28,14 @@ pub const HINT_KEYS: [&str; 5] = [
 
 pub(crate) fn extract_hints<T: ToHeaderMap>(headers: &T) -> Vec<(String, String)> {
     let mut hints = Vec::new();
+    if let Some(value_str) = headers.get(HINTS_KEY) {
+        value_str.split(',').for_each(|hint| {
+            let mut parts = hint.splitn(2, '=');
+            if let (Some(key), Some(value)) = (parts.next(), parts.next()) {
+                hints.push((key.trim().to_string(), value.trim().to_string()));
+            }
+        });
+    }
```

Comment:
If the header includes `HINTS_KEY`, it should be prioritized, allowing for an early return upon its discovery.

Reply:
You mean if we find `x-greptime-hints`, we should ignore keys starting with `x-greptime-hint-`?

Reply:
> You mean if we find `x-greptime-hints`, we should ignore keys starting with `x-greptime-hint-`?

Yes, always prefers `x-greptime-hints`.

Reply:
Ok, done in [e44f279](https://github.com/GreptimeTeam/greptimedb/pull/5194/commits/e44f27925582696c56d426606fcf897abb009205)